hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDDV2.scala

```diff
@@ -193,8 +192,8 @@ class HoodieMergeOnReadRDDV2(@transient sc: SparkContext,
             .withBaseFileOption(baseFileOption)
             .withPartitionPath(partitionPath)
             .withProps(properties)
-            .withDataSchema(HoodieSchema.parse(tableSchema.avroSchemaStr))
-            .withRequestedSchema(HoodieSchema.parse(requiredSchema.avroSchemaStr))
+            .withDataSchema(tableSchema.tableSchema)
+            .withRequestedSchema(requiredSchema.tableSchema)
```

Comment:
Should this be renamed from `tableSchema` to `schema` since it's not always table schema?